src/app/core/movies/edit-movie/movie-form.component.ts

```diff
@@ -0,0 +1,164 @@
+    private movieService: MoviesService,
+    private router: Router
+  ) {}
+  onCreateMovie() {
+    const id = String(Math.random());
+    if (!this.movieForm.valid) return;
+    const dto = { ...this.movieForm.value, id };
+    console.log(dto);
+
+    this.movieService
+      .addMovie(dto)
+      .pipe(takeUntil(this.temp$))
+      .subscribe({
+        next: (val) => this.router.navigateByUrl('/'),
+        error: (err) => console.log(err.message),
+      });
+    console.log(dto);
+  }
+  onEditMovie() {
+    if (!this.movieForm.valid) return;
+    const dto = { ...this.movieForm.value, id: this.movieId };
+    console.log(dto);
+
+    this.movieService
+      .updateMovie(dto, this.movieId)
+      .pipe(takeUntil(this.temp$))
+      .subscribe({
+        next: (val) => this.router.navigateByUrl('/'),
+        error: (err) => console.log(err.message),
+      });
+    console.log(dto);
+  }
+
+  temp$ = new Subject<void>();
+  currentMovie!: IMovie;
+  movieId!: number;
+  movieForm!: FormGroup;
+  edit: boolean = false;
+  create: boolean = false;
+  ngOnInit(): void {
+    this.movieForm = this.formBuilder.group({
+      title: [null, [Validators.required]],
+      rating: [null, [Validators.required]],
+      genre: new FormArray([new FormControl(null, Validators.required)]),
+    });
+    console.log(this.movieForm);
+    this.route.params
+      .pipe(
+        switchMap((val) => {
+          console.log(parseFloat(val['id']));
+          this.movieId = parseFloat(val['id']);
+          return this.movieService.getMovie(parseFloat(val['id'])).pipe(
+            catchError((err) => {
+              ///////////////////if this is catching why am i still getting error in console before log statement  where to put catch //////////
+              console.log(err);
+              return err;
+            }),
+            catchError((err) => {
+              ///////////////////if this is catching why am i still getting error in console before log statement  where to put catch//////////
+              console.log(err);
+              return err;
+            })
+          );
+        }),
+        takeUntil(this.temp$)
+      )
+      .subscribe({
+        ////////////// does not work with interface IMovie   could not fix
+        // next: (movie:IMovie) => {this.currentMovie = movie
+        next: (movie: any) => {
+          this.currentMovie = movie;
+          const genreArray = this.movieForm.get('genre') as FormArray;
+          genreArray.clear();
+
+          // Populate genre FormArray with movie.genre data
+          movie.genre.forEach((genre: string[]) => {
+            genreArray.push(new FormControl(genre, Validators.required));
+          });
+          this.movieForm.patchValue({
+            title: movie.title,
+            rating: movie.rating,
+          });
+        },
+        error: (err) => console.log(err.message),
+      });
+
+    ////////////////this does not work how to await subscription ? is this good practice?
```

Comment:
subscriptionebshi unda kna actionebi, anu subscribeshi gauwere is logika rac ginda ro moxdes da imis mere datrigerdeba rac response dagibrundeba